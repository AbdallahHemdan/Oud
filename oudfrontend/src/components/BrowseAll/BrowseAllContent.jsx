import React from "react";
import GenreCard from "./../GenreCard/GenreCard";

/**
 * @author AbdallahHemdan
 * @param {Object} items List of genres of the songs
 *
 * Functional component to display all genres
 */
const BrowseAllContent = ({ items }) => {
  return (
    <div className="wrapper" data-testid="first-wrapper">
      <div className="wrapper_section_2" data-testid="second-wrapper">
        <div className="cards" data-testid="cards-wrapper">
          {items.map((item, index) => {
            return <GenreCard item={item} key={index} />;
          })}
        </div>
      </div>
    </div>
  );
};

export default BrowseAllContent;
