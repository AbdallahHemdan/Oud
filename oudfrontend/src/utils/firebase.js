import firebase from "firebase";

const firebasConfig = {
  apiKey: "AIzaSyBZVUambAFEB-ywzAZTAPP-tSkbYXSEWc4",
  authDomain: "oud-notifications.firebaseapp.com",
  databaseURL: "https://oud-notifications.firebaseio.com",
  projectId: "oud-notifications",
  storageBucket: "oud-notifications.appspot.com",
  messagingSenderId: "681943816425",
  appId: "1:681943816425:web:9149a8834b2ff0f4623878"
};
// Initialize Firebase
firebase.initializeApp(firebasConfig);

export default firebase;
