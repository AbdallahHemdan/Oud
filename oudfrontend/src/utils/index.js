import axios from 'axios'
import { base } from '../config/environment'
const findByTestAttr = (component, val) => {
    return component.find(`[data-testid="${val}"]`);
}

function addToLikedSongs(id) {
    axios.put(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);

        })
        .catch(function (error) {
            console.log(error);
        });
}
function removeLikedSong(id) {
    axios.delete(`${base}/me/tracks/${id}`)
        .then(function (response) {
            console.log(response);
        })
        .catch(function (error) {
            console.log(error);
        });
}
export { findByTestAttr, addToLikedSongs, removeLikedSong }
