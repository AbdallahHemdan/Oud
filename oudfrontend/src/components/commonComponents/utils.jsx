import axios from "axios";
import { base } from "../../config/environment"
import {config} from "../../utils/auth"
/**
 * @file this file contains three functions used in Playlist, Album, LikedSongs 
 **/
export function addSong(playlistId, track){
  axios
  .post(`${base}/playlists/${playlistId}/${track}`, config)
  .then((response) => {
  })
  .catch((error) => {
  });
}