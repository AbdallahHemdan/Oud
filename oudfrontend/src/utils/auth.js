import jwtDecode from "jwt-decode";
import { getRequest } from "./requester";
import { base } from "./../config/environment";

function getToken() {
  return localStorage.getItem("accessToken");
}

const config = {
  headers: { authorization: `Bearer ${getToken()}` }
};

function Auth() {
  try {
    if (!getToken()) return null;
    let decoded = jwtDecode(getToken());
    console.log(decoded.id);
    return decoded.id;
  } catch {
    return null;
  }
}
function isLoggedIn() {
  return getToken() ? true : false;
}

function isArtist() {
  return getRequest(`${base}/me`)
    .then(response => {
      return response.data.role === "artist";
    })
    .catch(error => {
      console.log(error);
    });
}
const _config = config;
export { _config as config };
export const token = getToken();
const _Auth = Auth;
export { _Auth as Auth };
const _isLoggedIn = isLoggedIn;
export { _isLoggedIn as isLoggedIn };
const _isArtist = isArtist;
export { _isArtist as isArtist };
