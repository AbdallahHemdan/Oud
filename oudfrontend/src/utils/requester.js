import axios from "axios";
import { isLoggedIn } from "./../utils/auth";
const config = {
  headers: {
    authorization: `Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjVlOTA3ZGIwYTA2NDVmNDU4MTYwNzYwNiIsImlhdCI6MTU5MTYyMTQxOSwiZXhwIjoxNTk0MjEzNDE5fQ.fj3N3Pc89Pf_xlt7fGmXw1SINTecUB4-y3pihAAPjC8`,
    "Access-Control-Allow-Origin": "*"
  }
};
const getRequest = endpoint => {
  return axios.get(endpoint, config);
};
/**
 * Axios DELETE request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @returns {object}
 */
const deleteRequest = endpoint => {
  return axios.delete(endpoint, config);
};
/**
 * Axios PUT request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const putRequest = (endpoint, body = {}) => {
  return axios.put(endpoint, body, config);
};
/**
 * Axios POST request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const postRequest = (endpoint, body = {}) => {
  return axios.post(endpoint, body, config);
};
/**
 * Axios PATCH request
 * @function
 * @param {string} endpoint the required endpoint to implement an action
 * @param {object} body the request body
 * @returns {object}
 */
const patchRequest = (endpoint, body = {}) => {
  return axios.patch(endpoint, body, config);
};

// const isArtist = () => {
//   if (isLoggedIn()) {
//   }
// };
export { getRequest, deleteRequest, putRequest, postRequest, patchRequest };
