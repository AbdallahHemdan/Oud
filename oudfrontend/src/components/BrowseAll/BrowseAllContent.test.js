import React from 'react';
import BrowseAllContent from './BrowseAllContent';
import { shallow, mount } from 'enzyme';
import { findByTestAttr } from "./../../utils/index";
import checkPropTypes from 'check-prop-types';



const setup = (props = {}) => {
  const component = shallow(
    <BrowseAllContent {...props} />
  )
  return component;
}

let items = {
  items: [
    {
      _id: 1,
      name: "Recently played",
      icon: "https://previews.123rf.com/images/aratehortua/aratehortua2001/aratehortua200100022/138417489-vector-cartoon-low-poly-man-with-headphones-character.jpg"
    },
    {
      _id: 2,
      "name": "Recommended playlists",
      icon: "https://previews.123rf.com/images/belchonock/belchonock1801/belchonock180143844/94540927-musical-instruments-on-wooden-background.jpg"
    },
    {
      _id: 3,
      name: "New releases",
      icon: "https://media.altphotos.com/cache/images/2017/08/08/09/752/trumpet-outdoors-man.jpg"
    }
  ]
};

describe('Browse all content component', () => {
  let component;
  beforeEach(() => {
    component = setup(items);
  })

  it('Should render first wrapper in right way', () => {
    const wrapper = findByTestAttr(component, "first-wrapper");
    expect(wrapper.length).toBe(1);
  });

  it('Should render second wrapper in right way', () => {
    const wrapper = findByTestAttr(component, "second-wrapper");
    expect(wrapper.length).toBe(1);
  });

  it('Should render genres card wrapper in right way', () => {
    const wrapper = findByTestAttr(component, "cards-wrapper");
    expect(wrapper.length).toBe(1);
  });


  describe('snapshot test', () => {
    it('renders component correctly', () => {
      const wrapper = shallow(
        <BrowseAllContent items={items.items} />
      )
      expect(wrapper).toMatchSnapshot()
    });
  });

});