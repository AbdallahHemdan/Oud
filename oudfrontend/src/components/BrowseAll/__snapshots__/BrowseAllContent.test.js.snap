// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Browse all content component snapshot test renders component correctly 1`] = `ShallowWrapper {}`;
